### PR TITLE
MPI Operator plugin interface

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added Plugin contract for MPI

Dependent PR 
- https://github.com/flyteorg/flytepropeller/pull/314
- https://github.com/flyteorg/flytekit/pull/595
- https://github.com/flyteorg/flyteplugins/pull/196

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

https://github.com/flyteorg/flyte/issues/1543

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: protos/flyteidl/plugins/mpi.proto [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+// Custom proto for plugin that enables distributed training using https://github.com/kubeflow/mpi-operator
+message DistributedMPITrainingTask {
+  // number of worker spawned in the cluster for this job
+  int32 num_workers = 1;
+
+  // number of launcher replicas spawned in the cluster for this job
+  int32 num_launcher_replicas = 2;
+
+  // number of slots per worker used in hostfile.
```
Comment (katrogan) [inline protos/flyteidl/plugins/mpi.proto:15]: could you add more detail to this? what are slots used for?

File: protos/flyteidl/plugins/mpi.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+package flyteidl.plugins;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins";
+
+// Custom proto for plugin that enables distributed training using https://github.com/kubeflow/mpi-operator
+message DistributedMPITrainingTask {
+  // number of worker spawned in the cluster for this job
+  int32 num_workers = 1;
+
+  // number of launcher replicas spawned in the cluster for this job
```
Comment (katrogan) [inline protos/flyteidl/plugins/mpi.proto:12]: what is a launcher replica?

**Review by katrogan — APPROVED** [review]
thank you for adding comments!

**Comment by kumare3** [comment]
can you create an issue for this, then +1

**Review by EngHabu — APPROVED** [review]
